pkg/cmd/cli/describe/chaindescriber.go

```diff
@@ -189,3 +192,50 @@ func outputHelper(info, namespace string, singleNamespace bool) string {
 	}
 	return fmt.Sprintf("<%s %s>", namespace, info)
 }
+
+// DepthFirst implements stateful depth-first graph traversal.
+// Modifies behavior of visitor.DepthFirst to allow nodes to be visited multiple
+// times as long as they're not in the current stack
+type DepthFirst struct {
+	EdgeFilter func(graph.Edge) bool
+	Visit      func(u, v graph.Node)
+	stack      internal.NodeStack
+}
+
+// Walk performs a depth-first traversal of the graph g starting from the given node
+func (d *DepthFirst) Walk(g graph.Graph, from graph.Node, until func(graph.Node) bool) graph.Node {
+	return d.visit(g, from, until)
+}
+
+func (d *DepthFirst) visit(g graph.Graph, t graph.Node, until func(graph.Node) bool) graph.Node {
+	if until != nil && until(t) {
+		return t
+	}
+	d.stack.Push(t)
+	for _, n := range g.From(t) {
```

Comment:
Please force an ordering across these so that we're guaranteed to get repeatable output.